tests/net/socket/socketpair/src/main.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_test, CONFIG_NET_SOCKETS_LOG_LEVEL);
+
+#include <stdio.h>
+#include <string.h>
+#include <ztest_assert.h>
+
+#include <net/socket.h>
+
+//#include "../../socket_helpers.h"
+
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+#endif
+
+void test_socket_socketpair_open_close(void) {
+#if 0
+	int res;
+	int sv[2];
+
+	struct unstr {
+		unsigned u;
+		const char *s;
+	};
+	static const struct unstr address_family[] = {
+		{ AF_INET,  "AF_INET"  },
+		{ AF_INET6, "AF_INET6" },
+	};
+	static const struct unstr socket_type[] = {
+		{ SOCK_STREAM, "SOCK_STREAM" },
+		{ SOCK_DGRAM,  "SOCK_DGRAM"  },
+		{ SOCK_RAW,    "SOCK_RAW"  },
+	};
+	static char buf[128];
+
+	for(size_t i = 0; i < ARRAY_SIZE(address_family); ++i) {
+		for(size_t j = 0; j < ARRAY_SIZE(socket_type); ++j) {
+			res = socketpair(address_family[i], socket_type[j], 0, sv);
+
+			snprintf(buf, ARRAY_SIZE(buf), "socketpair did not return 0 or -1 but %d", res);
+			zassert_true(res == -1 || res == 0, buf);
+
+			snprintf(buf, ARRAY_SIZE(buf), "socketpair failed for address family %s and socket type %s", address_family[i], socket_type[j]);
+			zassert_equal(res, 0, buf);
+
+			res = close(sv[0]);
+
+			snprintf(buf, ARRAY_SIZE(buf), "close(sv[0]) failed for address family %s and socket type %s", address_family[i], socket_type[j]);
+			zassert_equal(res, 0, buf);
+
+			res = close(sv[1]);
+
+			snprintf(buf, ARRAY_SIZE(buf), "close(sv[1]) failed for address family %s and socket type %s", address_family[i], socket_type[j]);
+			zassert_equal(res, 0, buf);
+		}
+	}
+#endif
+	zassert_true(1, "1 was false");
+}
+
+void test_main(void)
+{
+	k_thread_system_pool_assign(k_current_get());
+
+	ztest_test_suite(socket_socketpair,
+			 ztest_user_unit_test(test_socket_socketpair_open_close));
+
```

Comment:
We need to add a test that sends/receives data and makes sure that the data is passed properly.